conans/client/remote_manager.py

```diff
@@ -39,6 +39,14 @@ def upload_recipe(self, conan_reference, remote, retry, retry_wait, ignore_delet
 
         t1 = time.time()
         export_folder = self._client_cache.export(conan_reference)
+
+        for f in (EXPORT_TGZ_NAME, EXPORT_SOURCES_TGZ_NAME):
+            tgz_path = os.path.join(export_folder, f)
+            if is_dirty(tgz_path):
+                self._output.warn("%s: Removing %s, marked as dirty" % (str(conan_reference), f))
+                os.remove(tgz_path)
+                clean_dirty(tgz_path)
+
```

Comment:
I can't see where the TGZ is reused in case it exists. In other words, why do you remove it (in case it is corrupted) and how it helps?

Reply:
Ok, now I see. It is in the _compress_recipe_file. A bit confusing, let me think.

Reply:
Shouldn't these dirty checks go into the `_compress_recipe_files` to invalidate the `export_tgz_path` and `sources_tgz_path` in case they are dirty?

Reply:
Tried to refactor it, but the solution inside ``_compress_recipe_files`` is more complex and ugly, because you need to remove the ".dirty" file from the gathered files